### DISCUSSION TITLE
Lesson 9 - Challenge

### DISCUSSION THREAD

**Discussion by Shiv2116** [Cyfrin/foundry-full-course-cu#148] [Q&A]
Has anyone been able to solve lesson 9 challenge?
i tried but was unable to solve . can anyone guide me at least what has to be done?

**Comment by palmcivet7** [comment]
Patrick left a comment here saying it is possible to solve it without deploying a contract: https://github.com/Cyfrin/foundry-full-course-f23/discussions/115#discussioncomment-6229388.

If you do try to solve the challenge by deploying a contract think about if there is anything from the challenge contract code you should include, if you need to adjust that code, and what happens to the NFT when the solveChallenge() function is successfully called.

**Reply by WillyEth** [reply]
I am stuck on this one, the NFT as you know gets minted but goes to the wrong spot.  Any hints.  Can I contact you show you my deployed contract?

**Comment by WillyEth** [comment]
@PatrickAlphaC any hints.  I deployed a contract, I believe I solved the answer, however the NFT reverts because its trying to send the mint to the deployed contract address.  I am either very close or going down the wrong road,  Spent 3 or 4 hours on this.

**Reply by ghost** [reply]
If you try to solve the challenge with a contract -- the challenge's contract will mint to that address. If you're sure you are computing `correctAnswer` correctly then there's something you need to implement in your solver contract, you should refer to the [EIP-721](https://eips.ethereum.org/EIPS/eip-721).

**Reply by WillyEth** [reply]
Got it.   I finally solved the challenge. It took awhile and a few hints from the community.

**Reply by Kel-VinTech** [reply]
hey how did you solve it?

**Reply by Niferu** [reply]
@Kel-VinTech 

None will tell you how to solve it as we avoid spoilers for community, you have to figure it out yourself. There are already a lot of hints in discussion tab :)
